atomic_reactor/plugins/pre_resolve_composes.py

```diff
@@ -319,6 +319,9 @@ def __init__(self, data, pulp_data, odcs_config, koji_tag=None, arches=None):
         self.modules = data.get('modules', [])
         self.pulp = {}
         self.arches = arches or []
+        if data.get('with_multilib', False):
+            if 'x86_64' in self.arches and 'i686' not in self.arches:
+                self.arches.append('i686')
```

Comment:
If i686 is added to self.arches now, it can be used to confirm the i686 that needs to be in pulp_repos during the check from 326-329.

If there isn't an i686 entry in pulp_repos, there's no way to do a multilib pulp build, because there wasn't an i686 entry in content_sets.yml and there's no way to guess what RPMs should be included in the i686 pulp compose. Currently we silently don't build the i686 pulp in that case, but I could change that to either log a warning or raise an error if you prefer.

Reply:
The `i686` packages should be available for building `x86_64` images. That's what the multilib story is all about. We're not building `i686` container images. 

If we handle adding `i686` in the render methods, the checks from 326-329 don't need changing.

Reply:
Okay, I misread the request.

Fixed.

Drastically extended the test cases to nail down every variation I could think of, but in a hopefully clearer manner.

Reply:
Fixed again, hopefully correctly.